### PR TITLE
[mini] small improvements around `MONO_IS_CALL` macro and `local-prop`

### PR DESCRIPTION
**Pull request by lewurm**
showed up in vtunes when running 
```
./mono/mini/mono-sgen --stats -O=all --aot=threads=1,asmonly ~/work/benchmarker/tests/fsharp/lib/bootstrap/4.0/FSharp.Compiler.dll
```

also, I think when `OP_DYN_CALL` was introduced, it wasn't added to the `MONO_IS_CALL` macro. @vargaz, what do you think?

### CODE REVIEW

File: mono/mini/mini.h [hunk 1]
```diff
@@ -484,7 +484,7 @@ enum {
 #define MONO_IS_STORE_MEMBASE(ins) (((ins)->opcode >= OP_STORE_MEMBASE_REG && (ins)->opcode <= OP_STOREV_MEMBASE) || ((ins)->opcode >= OP_ATOMIC_STORE_I1 && (ins)->opcode <= OP_ATOMIC_STORE_R8))
 #define MONO_IS_STORE_MEMINDEX(ins) (((ins)->opcode >= OP_STORE_MEMINDEX) && ((ins)->opcode <= OP_STORER8_MEMINDEX))
 
-#define MONO_IS_CALL(ins) (((ins->opcode >= OP_VOIDCALL) && (ins->opcode <= OP_VCALL2_MEMBASE)) || (ins->opcode == OP_TAILCALL))
+#define MONO_IS_CALL(ins) (((ins->opcode >= OP_VOIDCALL) && (ins->opcode <= OP_DYN_CALL)) || (ins->opcode == OP_TAILCALL))
```
Comment (vargaz) [inline mono/mini/mini.h]: IS_CALL means whenever the instruction is a MonoCallInst, and DYN_CALL is not.

File: mono/mini/mini.h [hunk 2]
```diff
@@ -484,7 +484,7 @@ enum {
 #define MONO_IS_STORE_MEMBASE(ins) (((ins)->opcode >= OP_STORE_MEMBASE_REG && (ins)->opcode <= OP_STOREV_MEMBASE) || ((ins)->opcode >= OP_ATOMIC_STORE_I1 && (ins)->opcode <= OP_ATOMIC_STORE_R8))
 #define MONO_IS_STORE_MEMINDEX(ins) (((ins)->opcode >= OP_STORE_MEMINDEX) && ((ins)->opcode <= OP_STORER8_MEMINDEX))
 
-#define MONO_IS_CALL(ins) (((ins->opcode >= OP_VOIDCALL) && (ins->opcode <= OP_VOIDCALL_MEMBASE)) || ((ins->opcode >= OP_FCALL) && (ins->opcode <= OP_FCALL_MEMBASE)) || ((ins->opcode >= OP_LCALL) && (ins->opcode <= OP_LCALL_MEMBASE)) || ((ins->opcode >= OP_VCALL) && (ins->opcode <= OP_VCALL_MEMBASE)) || ((ins->opcode >= OP_CALL) && (ins->opcode <= OP_CALL_MEMBASE)) || ((ins->opcode >= OP_VCALL2) && (ins->opcode <= OP_VCALL2_MEMBASE)) || (ins->opcode == OP_TAILCALL) || ((ins->opcode >= OP_RCALL) && (ins->opcode <= OP_RCALL_MEMBASE)))
+#define MONO_IS_CALL(ins) (((ins->opcode >= OP_VOIDCALL) && (ins->opcode <= OP_VCALL2_MEMBASE)) || (ins->opcode == OP_TAILCALL))
```
Comment (vargaz) [inline mono/mini/mini.h]: Looks ok, but need to make sure that all the call instructions are between those two.

File: mono/mini/mini.h (hunk 1 above)
Comment (lewurm) [inline mono/mini/mini.h]: thanks Zoltan, PR updated

**Comment by vargaz** [comment]
Looks ok, but needs a comment in mini-ops.h before OP_VOIDCALL which says that these opcodes need to stay together.

**Comment by lewurm** [comment]
added a comment to `mini-ops.h`

**Comment by vargaz** [comment]
LGTM.

**Comment by vargaz** [comment]
It would be nice to add a system for instruction metadata, i.e.
MINI_OP_FLAGS(OP_IADD, "int_add", IREG, IREG, IREG, MINI_OP_FLAG_NO_SIDE_EFFECT)

this could generate a byte array named op_flags, so you can do
op_flags [ins->opcode] & MINI_OP_FLAG_NO_SIDE_EFFECT to check instead of the horrible macros we have.

**Comment by kumpera** [comment]
Yes, please, lets get rid of all those error prone checks.

**Comment by lewurm** [comment]
yes, good point. I'll play around with that idea and create another PR.

**Comment by lewurm** [comment]
@monojenkins merge

**Pull request merged**